getItems.php

```diff
@@ -30,7 +30,12 @@ function getAllItems() {
                     <div class='itemBoxButtons'>
                     <button class='track-btn'>+ Track</button>";
             if (isset($_SESSION["loggedin"]) && $_SESSION["loggedin"] == TRUE) {
-                echo "<button class='details-btn'>View Details</button>";
+                $_SESSION["ITEMNAME"] = $row['itemName'];
+                $_SESSION["PRICE"] = $row['price'];
+                $_SESSION["ITEMID"] = $row['product_id'];
+                $_SESSION["IMG"] = $imgsrc;
+        
```

Comment:
Also, here your code actually overwrites the session variables for each product, so only the last variable will be stored. What you need to do is pass the product_id to the ViewDetails page via a get.

Reply:
whats the issue with using the session variable?